### PR TITLE
add vault's k8s authentication support

### PR DESCRIPTION
**Pull request by luanphantiki**
Hi @Lucretius , basically if we run the application on k8s, we can also use k8s authenticaion support. And here is my contributing that this may work. Hopefully it can help somehow.

Ref #9

### CODE REVIEW

**Comment by luanphantiki** [comment]
P/S: updated to WIP for testing

File: main.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ func main() {
 
 	for {
 		if snapshotter.TokenExpiration.Before(time.Now()) {
+			if c.VaultAuthMethod == "k8s" {
```
Comment (Lucretius) [inline main.go:49]: Won't this be overridden by the `snapshotter.SetClientTokenFromAppRole`?

Might want to convert this into a switch of some sort:

```
switch c.VaultAuthMethod {
case "k8s":
  snapshotter.SetClientTokenFromK8sAuth(c)
default:
  snapshotter.SetClientTokenFromAppRole(c)
}
```

**Review by Lucretius — COMMENTED** [review]
If you could modify the one suggested section, I will be happy to merge this in.

File: main.go (hunk 1 above)
Comment (luanphantiki) [reply main.go]: no that's alright, updated  this :+1: .

**Pull request merged**